docroot/modules/custom/layout_builder_custom/src/BannerBlockFormHandler.php

```diff
@@ -0,0 +1,548 @@
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public static function submitForm(array &$form, FormStateInterface $form_state) {
+    // Heading style radio selection.
+    $heading_style = $form_state->getValue('heading_style');
+    if ($heading_style) {
+      $form_state->setValue('layout_builder_style_headline_type', $heading_style);
+    }
+
+    // Gradient midpoint checkbox.
+    $adjust_gradient = $form_state->getValue('layout_builder_style_adjust_gradient_midpoint');
+    if (!$adjust_gradient) {
+      // Clear the gradient midpoint value if checkbox is unchecked.
+      $form_state->setValue('layout_builder_style_banner_gradient_midpoint', '');
+    }
+
+    $background_type = $form_state->getValue([
+      'settings',
+      'block_form',
+      'background_type',
+    ]);
+
+    if ($background_type) {
+      $form_media_selection = [
+        'settings',
+        'block_form',
+        'field_uiowa_banner_image',
+        'selection',
+      ];
+
+      $background_style = 'block_background_style_black';
+
+      if ($background_type === 'media') {
+        if ($form_state->getValue($form_media_selection)) {
+          $background_style = 'image';
+        }
+        // @todo Add feedback for user that they didn't upload an image.
+        // @todo Add validation to check whether image uploaded or not. See
+        // https://github.com/uiowa/uiowa/issues/5012
+      }
+      elseif ($background_type === 'color-pattern') {
+        // If a non-image background was selected, remove the reference.
+        // @todo Trigger file deletion if the media item is unused elsewhere. See
+        // https://github.com/uiowa/uiowa/issues/5013
+        $background_option = $form_state->getValue([
+          'settings',
+          'block_form',
+          'background_options',
+        ]);
+
+        if ($background_option) {
+          $background_style = $background_option;
+          $form_state->unsetValue($form_media_selection);
+        }
+      }
+
+      $form_state->setValue('layout_builder_style_background', $background_style);
+    }
+  }
+
+  /**
+   * Adds the gradient midpoint checkbox to the form.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  protected static function addGradientMidpointCheckbox(array &$form, FormStateInterface $form_state) {
+    // Check if gradient midpoint field has a value.
+    $has_midpoint_value = !empty($form['layout_builder_style_banner_gradient_midpoint']['#default_value']);
+
+    $form['layout_builder_style_adjust_gradient_midpoint'] = [
+      '#type' => 'checkbox',
+      '#title' => t('Adjust gradient midpoint'),
+      '#default_value' => $has_midpoint_value,
+      '#weight' => 94,
+      '#attributes' => [
+        'class' => ['adjust-gradient-checkbox'],
```

Comment:
It would be nice to see this as a BEM-ish style class name instead of something this bespoke.